wikimedia/wikipedia-ios
# Diffs - Connect Undo and Rollback buttons to API

**Phabricator:** 
https://phabricator.wikimedia.org/T335579

### Notes
This PR connects the undo and rollback buttons with the appropriate API calls. For the success banners, I just used our standard `WMFAlertManager` view, with the copy I observed from Android. I suggest we do tweaks to it at a later design review stage if needed. Since undo and rollbacks are edits, I am also parsing and presenting any MediaWiki error messages and presenting them in a panel, like we did with our recent custom blocked and page protection work.

### Test Steps
1. From any account (or anonymously), go to a recent diff revision on Test Wikipedia and tap through the undo flow. Confirm revision is undone and screen is refreshed to newest revision (and user-entered summary is also displayed in the TO: header).
2. From a non-admin Test Wikipedia account, go to an article's diff screen. Confirm Rollback is not an option. Then enter editor and make a few edits back-to-back on an article.
4. From a Test Wikipedia _admin_ account, go to one of those edits, confirm you can see rollback, and tap through the rollback flow. Confirm all user revisions from step 2 are reverted and screen is refreshed to newest revision.
5. Try doing both flows while in airplane mode. Confirm you see a "No internet connection" banner.
6. (Optional, we can lean on QA for this) Try the undo flow from a blocked account. Confirm you see an error panel and links are tappable.
